rust/gitxetcore/src/command/migrate.rs

```diff
@@ -0,0 +1,158 @@
+use std::path::PathBuf;
+use std::str::FromStr;
+
+use clap::Args;
+use tracing::error;
+
+use crate::config::XetConfig;
+use crate::errors::Result;
+use crate::git_integration::repo_migration::migrate_repo;
+use crate::git_integration::{clone_xet_repo, run_git_captured, run_git_passthrough, GitXetRepo};
+
+#[derive(Args, Debug, Clone)]
+pub struct MigrateArgs {
+    /// The URL of the git repository to import.
+    #[clap(long)]
+    pub src: String,
+
+    /// The URL of the Xet repository to initialize with the contents of the src repository.  
+    /// This repository must be a new Xet repo, or use --overwrite to replace all contents of
+    /// this repository with the source repo.
+    #[clap(long)]
+    pub dest: String,
+
+    /// The migrate command will error out if the remote repository is not empty.  
+    /// When this flag is passed, the contents of the remote repository will be
+    /// overwritten regardless of state.
+    #[clap(long)]
+    pub overwrite: bool,
+
+    /// Do not clean up the working directories after completion.
+    #[clap(long)]
+    pub no_cleanup: bool,
+
+    /// The directory to use to do all of the processing in (default: ~/.xet/migration).
+    #[clap(long)]
+    pub working_dir: Option<String>,
+}
+
+pub async fn migrate_command(config: XetConfig, args: &MigrateArgs) -> Result<()> {
+    let working_dir = {
+        if let Some(wd) = args.working_dir.as_ref() {
+            PathBuf::from_str(wd).unwrap()
+        } else {
+            let migration_tag = uuid::Uuid::now_v6(&[0; 6]).to_string();
+            config.xet_home.join("migration").join(migration_tag)
+        }
+    };
+
+    std::fs::create_dir_all(&working_dir)?;
```

Comment:
For dirs and files under xet_home, recommend to create using `config.permission.create_*` so there's no risk of root ownership when running the command under `sudo`.

Reply:
Good catch, will do!